kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/GCSToBQLoadRunnable.java

```diff
@@ -0,0 +1,232 @@
+package com.wepay.kafka.connect.bigquery;
+
+/*
+ * Copyright 2016 WePay, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.FormatOptions;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.LoadJobConfiguration;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.StorageException;
+
+import com.wepay.kafka.connect.bigquery.write.row.GCSToBQWriter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A Runnable that runs a GCS to BQ Load task.
+ *
+ * <p>This task goes through the given GCS bucket, and takes as many blobs as a single load job per
+ * table can handle (as defined here: https://cloud.google.com/bigquery/quotas#load_jobs) and runs
+ * those load jobs. Blobs are deleted (only) once a load job involving that blob succeeds.
+ */
+public class GCSToBQLoadRunnable implements Runnable {
+  private static final Logger logger = LoggerFactory.getLogger(GCSToBQLoadRunnable.class);
+
+  private final BigQuery bigQuery;
+  private final Bucket bucket;
+  private final Map<Job, List<Blob>> activeJobs;
+  private final Set<Blob> claimedBlobs;
+
+  // these numbers are intended to try to make this task not excede Google Cloud Quotas.
+  // see: https://cloud.google.com/bigquery/quotas#load_jobs
+  private static int FILE_LOAD_LIMIT = 10000;
+  // max number of files we can load in a single load job
```

Comment:
i think this comment and the one below are referring to the variables above them, which is a bit confusing.

Reply:
noted, I'll fix that.